copier/main.py

```diff
@@ -277,15 +278,21 @@ def _external_data(self) -> LazyDict:
         Files will only be parsed lazily on 1st access. This helps avoiding
         circular dependencies when the file name also comes from a variable.
         """
+        # Given those values are layzily rendered on 1st access then cached
+        # the phase value is irrelevant and could be misleading.
+        # As a consequence it is explicitely set to "undefined".
         return LazyDict(
             **{
-                name: lambda path=path: load_answersfile_data(
-                    self.dst_path, self._render_string(path)
-                )
+                name: lambda path=path: self._lazy_load_answers(path)
                 for name, path in self.template.external_data.items()
             }
         )
 
+    def _lazy_load_answers(self, path: str) -> AnyByStrDict:
+        """Lazy-Load external answers file."""
+        with Phase.use(Phase.UNDEFINED):
+            return load_answersfile_data(self.dst_path, self._render_string(path))
+
```

Comment:
I don't like the semantics of this method because it suggests that lazily loaded answers should always render the path with `_copier_phase = "undefined"` which isn't a general rule. I was thinking about a closure – see my other inline comment.